src/sonic-eventd/tools/rsyslog_volume_test.py

```diff
@@ -0,0 +1,61 @@
+import sys
+import subprocess
+import time
+import logging
+import argparse
+
+logging.basicConfig(
+    level=logging.INFO,
+    format="%(asctime)s [%(levelname)s] %(message)s",
+    handlers = [
+        logging.FileHandler("debug.log"),
+        logging.StreamHandler(sys.stdout)
+    ]
+)
+
+def read_events_from_file(count):
+    logging.info("Reading from file generated by events_tool")
+    lines = 0
+    with open('STDOUT', 'r') as file:
+        lines = file.readlines()
+        logging.info("Should receive {} events and got {} events\n".format(count, len(lines)))
+    assert len(lines) == count
+
+def start_tool():
+    logging.info("Starting events_tool\n")
+    proc = subprocess.Popen(["./events_tool", "-r"])
+    return proc
+
+def run_test(file, count):
+    # log messages to see if events have been received
+    tool_proc = start_tool()
+
+    time.sleep(2) # buffer for events_tool to startup
+
+    log_messages_cnt = 0
+    logging.info("Generating logger messages\n")
+    with open(file, 'r') as infile:
+        lines = infile.readlines()
+        log_messages_cnt = len(lines)
+        for line in lines:
+            line.rstrip()
+            command = "logger -p local0.notice -t {}\n".format(line)
+            for _ in range(count):
+                subprocess.run(command, shell=True, stdout=subprocess.PIPE)
+
+    time.sleep(2) # some buffer for all events to be published to file
+    read_events_from_file(count * log_messages_cnt)
+    tool_proc.terminate()
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-f", "--file", nargs='?', const='', default='', help="File containing log messages to be generated and read by rsyslog_plugin")
```

Comment:
If you set required = True, the python parser will fail on no arg.
Need not change. 
More a tip.